src/R/Components/Impl/InteractiveWorkflow/Implementation/RInteractiveEvaluator.cs

```diff
@@ -32,35 +33,49 @@ public sealed class RInteractiveEvaluator : IInteractiveEvaluator {
         private CarriageReturnProcessor _crProcessor;
         private int _terminalWidth = 80;
         private IInteractiveWindow _currentWindow;
+        private bool _isConnected;
 
         public IRHistory History { get; }
         public IRSession Session { get; }
 
         public RInteractiveEvaluator(IRSessionProvider sessionProvider, IRSession session, IRHistory history, IConnectionManager connections, ICoreShell coreShell, IRSettings settings, IConsole console) {
             History = history;
             Session = session;
-            Session.Output += SessionOnOutput;
-            Session.Disconnected += SessionOnDisconnected;
-            Session.BeforeRequest += SessionOnBeforeRequest;
-            Session.AfterRequest += SessionOnAfterRequest;
+
             _sessionProvider = sessionProvider;
             _connections = connections;
             _coreShell = coreShell;
             _settings = settings;
             _console = console;
             _evaluatorRequest = new CountdownDisposable();
-        }
 
-        public void Dispose() {
-            if (CurrentWindow != null) {
-                CurrentWindow.TextView.VisualElement.SizeChanged -= VisualElement_SizeChanged;
-            }
+            _disposableBag
+                .Add(() => Session.Output -= SessionOnOutput)
```

Comment:
Might as well put all the event deregistration in the single `{}` block that's already there below.